CMakeLists.txt

```diff
@@ -28,27 +26,64 @@ if(HIGHFIVE_PARALLEL_HDF5)
 set(HDF5_PREFER_PARALLEL TRUE)
 endif()
 
-find_package(HDF5 REQUIRED)
+find_package(HDF5 QUIET REQUIRED)
 
 set(Boost_NO_BOOST_CMAKE TRUE)
 
 
 if(USE_BOOST)
-    find_package(Boost 1.41 REQUIRED COMPONENTS system unit_test_framework)
-    add_definitions(-DH5_USE_BOOST )
+  find_package(Boost 1.41 QUIET REQUIRED COMPONENTS system unit_test_framework)
+  add_definitions(-DH5_USE_BOOST )
 endif()
 
 if(HIGHFIVE_PARALLEL_HDF5)
-    find_package(MPI)
+  find_package(MPI)
 endif()
 
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/highfive
 		DESTINATION ${INCLUDE_INSTALL_DIR})
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include ${HDF5_INCLUDE_DIRS} ${Boost_INCLUDE_DIR} ${MPI_C_INCLUDE_PATH})
+# cmake target install
+if(CMAKE_MAJOR_VERSION GREATER 2) # No export(EXPORT) in CMake 2
```

Comment:
Agreed for the separate file.

In any case, I think in the end it's not needed because I was doing something wrong initially (I should clean the CMakeCache.txt).
This is needed for facilitating FindPackage(HighFive) if you install it and not having to worry about much more (just adding HighFive to your target_link_libraries). 
In other words, this saves you from having to create a FindHighFive.cmake script, something you haven't faced because you are including HighFive as a submodule.

If you have such a strong opinion against it you can simply reject it, for me having to use an option to enable it it's almost the same thing as not having it at all.

And I don't think the cmake files it produces are polluting any more than pkgconfig does.